tests/test_livysession.py

```diff
@@ -64,15 +68,15 @@ def _next_session_response_post(self, *args):
         self.post_session_responses = self.post_session_responses[1:]
         return val
 
-    @staticmethod
-    def _create_session(kind=constants.SESSION_KIND_SPARK, session_id=-1,
-                        sql_created=False, http_client=None, spark_events=None):
-        if http_client is None:
-            http_client = MagicMock()
-        if spark_events is None:
-            spark_events = MagicMock()
+    def _create_session(self, kind=constants.SESSION_KIND_SPARK, session_id=-1, sql_created=False):
         ipython_display = MagicMock()
-        session = LivySession(http_client, {"kind": kind}, ipython_display, session_id, sql_created, spark_events)
+        session = LivySession(self.http_client, {"kind": kind}, ipython_display, session_id, sql_created, self.spark_events)
+        return session
+
+    def _create_session_with_fixed_get_response(self, get_session_json):
```

Comment:
All of these helper methods should be `staticmethod`s to ensure needless coupling.